openedx/core/lib/edx_api_utils.py

```diff
@@ -45,14 +42,7 @@ def get_edx_api_data(api_config, user, resource, resource_id=None, querystring=N
             return cached
 
     try:
-        jwt = get_id_token(user, api_config.OAUTH2_CLIENT_NAME)
```

Comment:
@ahsan-ul-haq instead of moving this code out and forcing it to be duplicated, why not allow an API client to be provided as a keyword argument? If it's present, use it, otherwise create one this way.